File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    sign = dict(zip(["0", "1"], [-1, 1]))
    n = len(numbers)
    for i in range(2 ** n):
        cnt = 0
        num = bin(i)[2:]
        m = len(num)
        if m < n:
            num = '0' * (n - m) + num
        for j in range(n):
            cnt += numbers[j] * sign[num[j]]
        if cnt == target:
            answer += 1
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cs
using System;

public class Solution 
{
    static int des,tar;
    static int answer = 0;
    public void search(int sum, int n, int[] arr)
    {
        if(n < des)
        {
            search(sum + arr[n], n+1, arr);
            search(sum - arr[n], n+1, arr);
        }
        else if(sum == tar)
            answer++;
            
    }
    public int solution(int[] numbers, int target) 
    {
        des = numbers.Length;
        tar = target;
        search(0,0,numbers);
        return answer;
    }
}